pkg/reconciler/pipelinerun/affinity_assistant_test.go

```diff
@@ -314,6 +328,76 @@ func TestCreateAndDeleteOfAffinityAssistantPerWorkspaceOrDisabled(t *testing.T)
 	}
 }
 
+func TestCreateOrUpdateAffinityAssistantsAndPVCs_Failure(t *testing.T) {
+	testCases := []struct {
+		name, failureType string
+		aaBehavior        aa.AffinityAssitantBehavior
+		expectedErr       error
+	}{{
+		name:        "affinity assistant creation failed - per workspace",
+		failureType: "statefulset",
+		aaBehavior:  aa.AffinityAssistantPerWorkspace,
+		expectedErr: fmt.Errorf("%w: [failed to create StatefulSet affinity-assistant-4cf1a1c468: error creating statefulsets]", ErrAffinityAssistantCreationFailed),
+	}, {
+		name:        "affinity assistant creation failed - per pipelinerun",
+		failureType: "statefulset",
+		aaBehavior:  aa.AffinityAssistantPerPipelineRun,
+		expectedErr: fmt.Errorf("%w: [failed to create StatefulSet affinity-assistant-426b306c50: error creating statefulsets]", ErrAffinityAssistantCreationFailed),
+	}, {
+		name:        "pvc creation failed - per workspace",
+		failureType: "pvc",
+		aaBehavior:  aa.AffinityAssistantPerWorkspace,
+		expectedErr: fmt.Errorf("%w: failed to create PVC pvc-b9eea16dce: error creating persistentvolumeclaims", ErrPvcCreationFailed),
+	}, {
+		name:        "pvc creation failed - disabled",
+		failureType: "pvc",
+		aaBehavior:  aa.AffinityAssistantDisabled,
+		expectedErr: fmt.Errorf("%w: failed to create PVC pvc-b9eea16dce: error creating persistentvolumeclaims", ErrPvcCreationFailed),
+	}}
+
+	for _, tc := range testCases {
+		ctx := context.Background()
+		kubeClientSet := fakek8s.NewSimpleClientset()
+		c := Reconciler{
+			KubeClientSet: kubeClientSet,
+			pvcHandler:    volumeclaim.NewPVCHandler(kubeClientSet, zap.NewExample().Sugar()),
+		}
+
+		switch tc.failureType {
+		case "pvc":
+			c.KubeClientSet.CoreV1().(*fake.FakeCoreV1).PrependReactor("create", "persistentvolumeclaims",
+				func(action testing2.Action) (handled bool, ret runtime.Object, err error) {
+					return true, &corev1.PersistentVolumeClaim{}, errors.New("error creating persistentvolumeclaims")
```

Comment:
nit: could the switch statement be replaced with:
```
c.KubeClientSet.CoreV1().(*fake.FakeCoreV1).PrependReactor("create", tc.failureType,
				func(action testing2.Action) (handled bool, ret runtime.Object, err error) {
					return true, <something implementing runtime.Object>, errors.New("error creating %s", tc.failureType)
```

(I don't think the type of object returned in the nested func matters, but I could be wrong)

Reply:
hmmm, seems like the type need to match otherwise:

```
--- FAIL: TestCreateOrUpdateAffinityAssistantsAndPVCs_Failure (0.00s)
panic: interface conversion: runtime.Object is *v1.StatefulSet, not *v1.PersistentVolumeClaim [recovered]
	panic: interface conversion: runtime.Object is *v1.StatefulSet, not *v1.PersistentVolumeClaim
```